src/app/CommandHandlerDispatcher.cpp

```diff
@@ -0,0 +1,128 @@
+#include "CommandHandlerDispatcher.h"
+#include "InteractionModelEngine.h"
+#include "messaging/ExchangeContext.h"
+
+namespace chip {
+namespace app {
+using Status = Protocols::InteractionModel::Status;
+
+CHIP_ERROR CommandHandlerDispatcher::OnMessageReceived(Messaging::ExchangeContext * apExchangeContext,
+                                                       const PayloadHeader & aPayloadHeader, System::PacketBufferHandle && aPayload)
+{
+    CHIP_ERROR err                           = CHIP_NO_ERROR;
+    bool sendStatusResponseWithInvalidAction = false;
+
+    if (mState == State::AwaitingResponse && aPayloadHeader.HasMessageType(Protocols::InteractionModel::MsgType::StatusResponse))
+    {
+        CHIP_ERROR statusError = CHIP_NO_ERROR;
+        err                    = StatusResponse::ProcessStatusResponse(std::move(aPayload), statusError);
+        VerifyOrExit(err == CHIP_NO_ERROR, sendStatusResponseWithInvalidAction = true);
+        statusError = err;
+        VerifyOrExit(err == CHIP_NO_ERROR, sendStatusResponseWithInvalidAction = true);
+
+        err = SendCommandResponse();
+        VerifyOrExit(err == CHIP_NO_ERROR, sendStatusResponseWithInvalidAction = true);
```

Comment:
Why would we send a status with InvalidAction if _we_ failed at something?  That doesn't make sense.

Reply:
What should we send here. There isn't anything in the spec on what to do here

Reply:
Either nothing or a generic Failure status, I would think.